include/power/power_state.h

```diff
@@ -13,18 +13,153 @@
 extern "C" {
 #endif
 
-#define PM_STATE_RUNTIME_ACTIVE  BIT(0)
-#define PM_STATE_RUNTIME_IDLE    BIT(1)
-#define PM_STATE_SUSPEND_TO_IDLE BIT(2)
-#define PM_STATE_STANDBY         BIT(3)
-#define PM_STATE_SUSPEND_TO_RAM  BIT(4)
-#define PM_STATE_SUSPEND_TO_DISK BIT(5)
-#define PM_STATE_SOFT_OFF        BIT(6)
+/**
+ * @brief Runtime active state
+ *
+ * During runtime active state, the system is awake and running.
+ * In simple terms, the system is in a full running state.
+ */
+#define PM_STATE_RUNTIME_ACTIVE        BIT(0)
+
+/**
+ * @brief Runtime idle state
+ *
+ * Runtime idle is a system sleep state in which all of the cores
+ * enter deepest possible idle state and wait for interrupts, no
+ * requirements for the devices, leaving them at the states where
+ * they are.
```

Comment:
Could this be:

> Runtime idle is the state entered when the only work to be done is the kernel idle thread.  That should be a system sleep state in which ....

?  I'm not sure that's entirely correct, as there are a very small number of places in the code where the current thread tries to go into idle mode without being a reschedule point, but I think it's worth describing this in terms of the expected behavior of the idle thread.

Reply:
> as there are a very small number of places in the code where the current thread tries to go into idle mode without being a reschedule point

yes, I think we can enter that mode even not in idle thread.